### DISCUSSION TITLE
How can i use "precomputed" HDBSCAN in BerTopic?

### DISCUSSION THREAD

**Discussion by shadabmeymandi** [MaartenGr/BERTopic#2202] [Q&A]
when i use metric="precomputed" in my HDBSCAN and then run my BerTopic model i face the error:

ValueError: operands could not be broadcast together with shapes (5,) (5,498) 

how can i fix this?

**Comment by MaartenGr** [comment]
It's difficult to say without more specifics. Could you create an issue and fill in the necessary information?

**Reply by shadabmeymandi** [reply]
umap_model = umap.UMAP(n_neighbors=15, n_components=24, min_dist=0.0, metric='cosine', random_state=100)
embedding_model = SentenceTransformer('sentence-transformers/paraphrase-multilingual-mpnet-base-v2')
hdbscan_model =  HDBSCAN(metric='euclidean',min_cluster_size=10, cluster_selection_method='eom', prediction_data=True)
model = BERTopic( hdbscan_model=hdbscan_model,embedding_model=embedding_model,umap_model=umap_model, language="english", calculate_probabilities=True)
topics, probabilities =model.fit_transform(sentecnes)

as you know when i run the above code there is no problem and every thing is ok , but i must use the hdbscan as below:

hdbscan_model =  HDBSCAN(metric='precomputed',min_cluster_size=10, cluster_selection_method='eom', prediction_data=True)
cluster_labels = hdbscan_model.fit_predict(distances)

because i have a pairwaise matrix "distances[ ]" as the distances between my embeddings. 
with this metric='precomputed' i cant't run my bertopic model and the error is : 

ValueError: operands could not be broadcast together with shapes (24,) (24,489)
